packager/media/formats/mp2t/ts_segmenter.cc

```diff
@@ -169,30 +166,41 @@ Status TsSegmenter::WritePesPacketsToFile() {
 }
 
 Status TsSegmenter::FinalizeSegment(uint64_t start_timestamp,
-                                    uint64_t duration) {
+                                    uint64_t duration,
+                                    uint64_t segment_index) {
   if (!pes_packet_generator_->Flush()) {
-    return Status(error::MUXER_FAILURE,
-                  "Failed to flush PesPacketGenerator.");
+    return Status(error::MUXER_FAILURE, "Failed to flush PesPacketGenerator.");
   }
   Status status = WritePesPacketsToFile();
   if (!status.ok())
     return status;
 
-  // This method may be called from Finalize() so ts_writer_file_opened_ could
+  // This method may be called from Finalize() so ts_writer_buffer_initialized_ could
   // be false.
-  if (ts_writer_file_opened_) {
+  if (ts_writer_buffer_initialized_) {
+    std::string segment_name_segment_index =
+        GetSegmentName(muxer_options_.segment_template, next_pts_,
+                       segment_index - 1, muxer_options_.bandwidth);
+
+    current_segment_path_ = segment_name_segment_index;
+    if (!ts_writer_->CreateFileAndFlushBuffer(segment_name_segment_index)) {
```

Comment:
Can we call the function: `WriteToFile`?

Reply:
Sure.